File: apps/mobile/android/shared/src/commonTest/kotlin/cc/edecan/shared/ChatSecretRedactionTest.kt
package cc.edecan.shared

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse

class ChatSecretRedactionTest {
    @Test
    fun redactsOpenAiProjectKeyBeforeRendering() {
        val secret = "sk-proj-example-secret-1234567890"
        val result = ChatSecretRedaction.redact("Configura OpenAI con $secret")

        assertEquals("Configura OpenAI con [credencial protegida]", result)
        assertFalse(result.contains(secret))
    }

    @Test
    fun leavesOrdinaryConversationUntouched() {
        val text = "Crea una imagen de una chica en un balcón."
        assertEquals(text, ChatSecretRedaction.redact(text))
    }
}

File: apps/web/chat-secret-redaction.test.mjs
import assert from "node:assert/strict";
import { readFile } from "node:fs/promises";
import test from "node:test";

import { redactChatSecrets } from "./src/lib/chat-secret-redaction.ts";

test("la burbuja optimista nunca pinta una API key", () => {
  const secret = "sk-proj-example-secret-1234567890";
  const result = redactChatSecrets(`Configura OpenAI con ${secret}`);

  assert.equal(result, "Configura OpenAI con [credencial protegida]");
  assert.ok(!result.includes(secret));
});

test("el chat muestra la versión redactada pero conserva el texto original para el request", async () => {
  const source = await readFile(new URL("./src/app/(app)/app/page.tsx", import.meta.url), "utf8");

  assert.match(source, /text:\s*redactChatSecrets\(text\)/);
  assert.match(source, /sendMessageStream\([\s\S]*?activeId,[\s\S]*?text,/);
});
